From 808de7ed4f68926af2a940b340695d580bcc0565 Mon Sep 17 00:00:00 2001
From: Nikita Yushchenko <nikita.yoush@cogentembedded.com>
Date: Fri, 28 Jan 2022 17:45:20 +0300
Subject: [PATCH 020/151] dts: renesas: vc4: clenup

- merge into single file (no need to have cpu.dtsi when there is no
  separate cpu board)

- sort alphabetically

- remove definitions stale from other board(s)

- define scif-clk pinmix in scif-clk node

- add vcc-supply to EEPROM node, to avoid warning in logs

Keep for now two settings that are wrong but somehow workaround boot
hangs. Namely, disable &gpio4 and don't enable &pciec1.

Signed-off-by: Nikita Yushchenko <nikita.yoush@cogentembedded.com>
---
 .../boot/dts/renesas/r8a779f0-vc4-cpu.dtsi    | 233 ------------------
 arch/arm64/boot/dts/renesas/r8a779f0-vc4.dts  | 219 +++++++++++++---
 2 files changed, 182 insertions(+), 270 deletions(-)
 delete mode 100644 arch/arm64/boot/dts/renesas/r8a779f0-vc4-cpu.dtsi

diff --git a/arch/arm64/boot/dts/renesas/r8a779f0-vc4-cpu.dtsi b/arch/arm64/boot/dts/renesas/r8a779f0-vc4-cpu.dtsi
deleted file mode 100644
index 6b111bcfe1f7..000000000000
--- a/arch/arm64/boot/dts/renesas/r8a779f0-vc4-cpu.dtsi
+++ /dev/null
@@ -1,233 +0,0 @@
-// SPDX-License-Identifier: GPL-2.0
-/*
- * Device Tree Source for the VC4 board
- *
- * Copyright (C) 2021 Renesas Electronics Corp.
- */
-
-#include <dt-bindings/gpio/gpio.h>
-#include "r8a779f0.dtsi"
-
-/ {
-	model = "Renesas VC4 board";
-	compatible = "renesas,vc4", "renesas,r8a779f0";
-
-	aliases {
-		serial0 = &scif3;
-		serial1 = &scif0;
-	};
-
-	chosen {
-		stdout-path = "serial0:115200n8";
-	};
-
-	memory@48000000 {
-		device_type = "memory";
-		/* first 128MB is reserved for secure area. */
-		reg = <0x0 0x48000000 0x0 0x78000000>;
-	};
-
-/*
-	memory@480000000 {
-		device_type = "memory";
-		reg = <0x4 0x80000000 0x0 0x80000000>;
-	};
-*/
-
-	reg_1p8v: regulator-1p8v {
-		compatible = "regulator-fixed";
-		regulator-name = "fixed-1.8V";
-		regulator-min-microvolt = <1800000>;
-		regulator-max-microvolt = <1800000>;
-		regulator-boot-on;
-		regulator-always-on;
-	};
-
-	reg_3p3v: regulator-3p3v {
-		compatible = "regulator-fixed";
-		regulator-name = "fixed-3.3V";
-		regulator-min-microvolt = <3300000>;
-		regulator-max-microvolt = <3300000>;
-		regulator-boot-on;
-		regulator-always-on;
-	};
-
-	vcc_sdhi: regulator-vcc-sdhi {
-		compatible = "regulator-fixed";
-
-		regulator-name = "SDHI Vcc";
-		regulator-min-microvolt = <3300000>;
-		regulator-max-microvolt = <3300000>;
-
-		gpio = <&gpio1 24 GPIO_ACTIVE_HIGH>;
-		enable-active-high;
-	};
-};
-
-&extal_clk {
-	clock-frequency = <20000000>;
-};
-
-&extalr_clk {
-	clock-frequency = <32768>;
-};
-
-&i2c1 {
-	pinctrl-0 = <&i2c1_pins>;
-	pinctrl-names = "default";
-
-	status = "okay";
-	clock-frequency = <400000>;
-
-	isl1208_rtc: rtc@6f {
-		compatible = "isil,isl1208";
-		reg = <0x6f>;
-	};
-};
-
-&i2c1 {
-	pinctrl-0 = <&i2c1_pins>;
-	pinctrl-names = "default";
-
-	status = "okay";
-	clock-frequency = <400000>;
-
-	isl1208_rtc: rtc@6f {
-		compatible = "isil,isl1208";
-		reg = <0x6f>;
-	};
-};
-
-&i2c4 {
-	pinctrl-0 = <&i2c4_pins>;
-	pinctrl-names = "default";
-
-	status = "okay";
-	clock-frequency = <400000>;
-
-	eeprom@50 {
-		compatible = "st,24c16", "atmel,24c16";
-		label = "board-data";
-		reg = <0x50>;
-		pagesize = <8>;
-	};
-
-	fpga4000@71 {
-		compatible = "renesas,vc4-fpga4000";
-		reg = <0x71>;
-	};
-};
-
-&mmc0 {
-	pinctrl-0 = <&sd_pins>;
-	pinctrl-1 = <&sd_pins>;
-	pinctrl-names = "default", "state_uhs";
-
-	vmmc-supply = <&vcc_sdhi>;
-	vqmmc-supply = <&vcc_sdhi>;
-	cd-gpios = <&gpio1 23 GPIO_ACTIVE_LOW>;
-	bus-width = <4>;
-	status = "okay";
-};
-
-&pfc {
-	pinctrl-0 = <&scif_clk_pins>;
-	pinctrl-names = "default";
-
-	i2c1_pins: i2c1 {
-		groups = "i2c1";
-		function = "i2c1";
-	};
-
-	i2c1_pins: i2c1 {
-		groups = "i2c1";
-		function = "i2c1";
-	};
-
-	i2c4_pins: i2c4 {
-		groups = "i2c4";
-		function = "i2c4";
-	};
-
-	mmc_pins: mmc {
-		groups = "mmc_data8", "mmc_ctrl", "mmc_ds";
-		function = "mmc";
-		power-source = <1800>;
-	};
-
-	sd_pins: sd {
-		groups = "mmc_data4", "mmc_ctrl";
-		function = "mmc";
-		power-source = <3300>;
-	};
-
-	scif0_pins: scif0 {
-		groups = "scif0_data", "scif0_ctrl";
-		function = "scif0";
-	};
-
-	scif3_pins: scif3 {
-		groups = "scif3_data", "scif3_ctrl";
-		function = "scif3";
-	};
-
-	hscif0_pins: hscif0 {
-		groups = "hscif0_data", "hscif0_ctrl";
-		function = "hscif0";
-	};
-
-	hscif1_pins: hscif1 {
-		groups = "hscif1_data", "hscif1_ctrl";
-		function = "hscif1";
-	};
-
-	scif_clk_pins: scif_clk {
-		groups = "scif_clk";
-		function = "scif_clk";
-	};
-};
-
-&rwdt {
-	timeout-sec = <60>;
-	status = "okay";
-};
-
-&scif0 {
-	pinctrl-0 = <&scif0_pins>;
-	pinctrl-names = "default";
-
-	uart-has-rtscts;
-	/* Please use exclusively to the hscif1 node */
-	status = "okay";
-};
-
-&scif3 {
-	pinctrl-0 = <&scif3_pins>;
-	pinctrl-names = "default";
-
-	uart-has-rtscts;
-	/* Please use exclusively to the hscif0 node */
-	status = "okay";
-};
-
-&hscif0 {
-	pinctrl-0 = <&hscif0_pins>;
-	pinctrl-names = "default";
-
-	uart-has-rtscts;
-	/* Please use exclusively to the scif3 node */
-	status = "disabled";
-};
-
-&hscif1 {
-	pinctrl-0 = <&hscif1_pins>;
-	pinctrl-names = "default";
-
-	uart-has-rtscts;
-	/* Please use exclusively to the scif0 node */
-	status = "disabled";
-};
-
-&scif_clk {
-	clock-frequency = <24000000>;
-};
diff --git a/arch/arm64/boot/dts/renesas/r8a779f0-vc4.dts b/arch/arm64/boot/dts/renesas/r8a779f0-vc4.dts
index 95562d5d6d9d..dbb33802dcb8 100644
--- a/arch/arm64/boot/dts/renesas/r8a779f0-vc4.dts
+++ b/arch/arm64/boot/dts/renesas/r8a779f0-vc4.dts
@@ -6,18 +6,166 @@
  */
 
 /dts-v1/;
-#include "r8a779f0-vc4-cpu.dtsi"
+
+#include "r8a779f0.dtsi"
+#include <dt-bindings/gpio/gpio.h>
 
 / {
-	model = "Renesas VC4 vehicle computer board based on R-Car S4 (r8a779f0)";
-	compatible = "renesas,vc4-cpu", "renesas,r8a779f0";
+	model = "Renesas VC4 board based on r8a779f0";
+	compatible = "renesas,vc4", "renesas,r8a779f0";
 
 	aliases {
+		serial0 = &scif3;
+		serial1 = &scif0;
 		eth0 = &rswitch;
-		};
+	};
+
+	chosen {
+		stdout-path = "serial0:115200n8";
+	};
+
+	memory@48000000 {
+		device_type = "memory";
+		/* first 128MB is reserved for secure area. */
+		reg = <0x0 0x48000000 0x0 0x78000000>;
+	};
+
+	reg_1p8v: regulator-1p8v {
+		compatible = "regulator-fixed";
+		regulator-name = "fixed-1.8V";
+		regulator-min-microvolt = <1800000>;
+		regulator-max-microvolt = <1800000>;
+		regulator-boot-on;
+		regulator-always-on;
+	};
+
+	reg_3p3v: regulator-3p3v {
+		compatible = "regulator-fixed";
+		regulator-name = "fixed-3.3V";
+		regulator-min-microvolt = <3300000>;
+		regulator-max-microvolt = <3300000>;
+		regulator-boot-on;
+		regulator-always-on;
+	};
+
+	vcc_sdhi: regulator-vcc-sdhi {
+		compatible = "regulator-fixed";
+
+		regulator-name = "SDHI Vcc";
+		regulator-min-microvolt = <3300000>;
+		regulator-max-microvolt = <3300000>;
+
+		gpio = <&gpio1 24 GPIO_ACTIVE_HIGH>;
+		enable-active-high;
+	};
+};
+
+&extal_clk {
+	clock-frequency = <20000000>;
+};
+
+&extalr_clk {
+	clock-frequency = <32768>;
+};
+
+// Allowing &gpio4 to initialize causes boot hang
+&gpio4 {
+	status = "disable";
+};
+
+&i2c1 {
+	status = "okay";
+
+	pinctrl-0 = <&i2c1_pins>;
+	pinctrl-names = "default";
+
+	clock-frequency = <400000>;
+
+	rtc@6f {
+		compatible = "isil,isl1208";
+		reg = <0x6f>;
+	};
+};
+
+&i2c4 {
+	status = "okay";
+
+	pinctrl-0 = <&i2c4_pins>;
+	pinctrl-names = "default";
+
+	clock-frequency = <400000>;
+
+	eeprom@50 {
+		compatible = "st,24c16", "atmel,24c16";
+		reg = <0x50>;
+
+		vcc-supply = <&reg_3p3v>;
+
+		label = "board-data";
+		pagesize = <8>;
+	};
+
+	fpga4000@71 {
+		compatible = "renesas,vc4-fpga4000";
+		reg = <0x71>;
+	};
+};
+
+&mmc0 {
+	status = "okay";
+
+	pinctrl-0 = <&sd_pins>;
+	pinctrl-1 = <&sd_pins>;
+	pinctrl-names = "default", "state_uhs";
+
+	vmmc-supply = <&vcc_sdhi>;
+	vqmmc-supply = <&vcc_sdhi>;
+	cd-gpios = <&gpio1 23 GPIO_ACTIVE_LOW>;
+	bus-width = <4>;
 };
 
 &pfc {
+	i2c1_pins: i2c1 {
+		groups = "i2c1";
+		function = "i2c1";
+	};
+
+	i2c4_pins: i2c4 {
+		groups = "i2c4";
+		function = "i2c4";
+	};
+
+	pcie0_pins: pcie0 {
+		groups = "pcie0_clkreq_n";
+		function = "pcie";
+	};
+
+	pcie1_pins: pcie1 {
+		groups = "pcie1_clkreq_n";
+		function = "pcie";
+	};
+
+	scif0_pins: scif0 {
+		groups = "scif0_data", "scif0_ctrl";
+		function = "scif0";
+	};
+
+	scif3_pins: scif3 {
+		groups = "scif3_data", "scif3_ctrl";
+		function = "scif3";
+	};
+
+	scif_clk_pins: scif_clk {
+		groups = "scif_clk";
+		function = "scif_clk";
+	};
+
+	sd_pins: sd {
+		groups = "mmc_data4", "mmc_ctrl";
+		function = "mmc";
+		power-source = <3300>;
+	};
+
 	tsn0_pins: tsn0 {
 		mux {
 			groups = "tsn0_link", "tsn0_mdio";
@@ -30,25 +178,30 @@
 			drive-strength = <12>;
 		};
 	};
+};
 
-	pcie0_pins: pcie0 {
-		groups = "pcie0_clkreq_n";
-		function = "pcie";
-	};
+&pciec0 {
+	status = "okay";
 
-	pcie1_pins: pcie1 {
-		groups = "pcie1_clkreq_n";
-		function = "pcie";
-	};
+	pinctrl-0 = <&pcie0_pins>;
+	pinctrl-names = "default";
+};
 
-	taud0_pwm_pins: taud0_pwm {
-		groups = "taud0_pwm1", "taud0_pwm2";
-		function = "taud0_pwm";
-	};
+&pciec1 {
+	// Exists on the board, but enabling it currently causes boot hang
+	//status = "okay";
+
+	pinctrl-0 = <&pcie1_pins>;
+	pinctrl-names = "default";
+};
+
+&pcie_bus_clk {
+	clock-frequency = <100000000>;
 };
 
 &rswitch {
 	status = "okay";
+
 	#address-cells = <1>;
 	#size-cells = <0>;
 
@@ -89,35 +242,27 @@
 	};
 };
 
-&pcie_bus_clk {
-	clock-frequency = <100000000>;
-};
-
-&pciec0 {
+&scif0 {
 	status = "okay";
-	pinctrl-0 = <&pcie0_pins>;
-	pinctrl-names = "default";
-};
 
-&pciec0_ep {
-	status = "disabled";
-	pinctrl-0 = <&pcie0_pins>;
+	pinctrl-0 = <&scif0_pins>;
 	pinctrl-names = "default";
-};
 
-&pciec1 {
-	status = "disabled";
-	pinctrl-0 = <&pcie1_pins>;
-	pinctrl-names = "default";
+	uart-has-rtscts;
 };
 
-&tau_pwm0 {
-	pinctrl-0 = <&taud0_pwm_pins>;
+&scif3 {
+	status = "okay";
+
+	pinctrl-0 = <&scif3_pins>;
 	pinctrl-names = "default";
 
-	status = "disable";
+	uart-has-rtscts;
 };
 
-&gpio4 {
-	status = "disable";
+&scif_clk {
+	pinctrl-0 = <&scif_clk_pins>;
+	pinctrl-names = "default";
+
+	clock-frequency = <24000000>;
 };
-- 
2.17.1

